src/index.ts

```diff
@@ -146,15 +115,11 @@ export const createRouter = <
     replace?: boolean | undefined;
     target?: React.HTMLAttributeAnchorTarget | undefined;
   }) => {
-    const active = useSubscription(
-      React.useMemo(
-        () => ({
-          getCurrentValue: () => href === currentLocation.url,
-          subscribe,
-        }),
-        [href],
-      ),
-    );
+    const location = useLocation();
+    // TODO: explode/implode `href` in order to guarantee the search params order
```

Comment:
`encodeSearch` and `decodeSearch` have deterministic params order already 🙂

Reply:
Yeah, but `href` can be of free-form, right?

Reply:
Ah yes, with the addition of `unsafeNavigate` and `unsafeReplace`.